packages/esm-patient-search-app/src/compact-patient-search-extension/index.tsx

```diff
@@ -0,0 +1,45 @@
+import React, { useCallback, useState, useMemo } from 'react';
+import PatientSearch from '../compact-patient-search/patient-search.component';
+import PatientSearchBar from './patient-search-bar.component';
+import styles from './compact-patient-search.scss';
+import { SearchedPatient } from '../types';
+import debounce from 'lodash-es/debounce';
+
+interface CompactPatientSearchProps {
+  initialSearchTerm: string;
+  selectPatientAction?: (patient: SearchedPatient) => undefined;
+  buttonProps?: Object;
+}
+
+const CompactPatientSearchComponent: React.FC<CompactPatientSearchProps> = ({
+  selectPatientAction,
+  initialSearchTerm = '',
+  buttonProps,
+}) => {
+  const [searchTerm, setSearchTerm] = useState(initialSearchTerm);
+  const handleSearchTerm = debounce((val) => setSearchTerm(val), 300);
+  const showSearchResults = useMemo(() => !!searchTerm.trim(), [searchTerm]);
```

Comment:
I think this unprotected trim is causing errors for me. Can you add an optional chaining here?

Reply:
Actually the prop `initialSearchTerm` was undefined, that caused the error. I gave it a default value in here, and it works as needed. I'll add an optional chaining. No issues.
Thanks!